### PR TITLE
Fix template lookup description

### PR DESCRIPTION
**Pull request by jpmahowald**
##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

##### SUMMARY
Previously, template lookup description was documented  as "mostly a noop".
Now,  describe that it returns template content with substituted variables.

##### COMPONENT NAME
template

### CODE REVIEW

File: lib/ansible/plugins/lookup/template.py [hunk 1]
```diff
@@ -11,7 +11,8 @@
     version_added: "0.9"
     short_description: retrieve contents of file after templating with Jinja2
     description:
-      - this is mostly a noop, to be used as a with_list loop when you do not want the content transformed in any way.
+      - Retrieves the content of a template after substituting variables.
+      - Same content as using the template module, but returns a string instead of writing an output file.
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/template.py]: I would rephrase as `Returns a string(s) resulting from processing the template file(s) provided.`

**Review by bcoca — DISMISSED** [review]

**Comment by acozine** [comment]
Thanks @jpmahowald for contributing to the Ansible documentation!
